### PR TITLE
LP-154 Add Makefile and Sonar config

### PR DESCRIPTION
**Pull request by mwestacott**
https://companieshouse.atlassian.net/browse/LP-154

* Unused Makefile targets are currently commented out

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	rm -rf ./build-*
+	rm -rf ./dist
+	rm -f ./build.log
+
+.PHONY: build
+build:
+	npm ci
+	npm run build
+
+# .PHONY: lint
```
Comment (mwestacott) [inline Makefile:19]: Thoughts on this?

I copied the OE Web Makefile and made changes, then commented out the targets that aren't yet applicable (until we complete the other tasks).

Should I leave them commented out (easy to add back in later/soon) or just delete for now?

Comment (markpit) [reply Makefile:19]: you can uncomment them if they will be needed

File: README.md [hunk 2]
```diff
@@ -9,4 +9,11 @@ Requires >= Node v20
 2. run 'npm run build' to compile
 3. run 'npm run start' to start the service
 
-use localhost:3000/limited-partnerships/start to view the test page
\ No newline at end of file
+use localhost:3000/limited-partnerships/start to view the test page
+
+## Frontend Technologies and Utils
+
+- [NodeJS](https://nodejs.org/)
```
Comment (mattch1) [inline README.md:16]: I assume we'll add to this as we go along, docker instructions etc.

**Review by mattch1 — APPROVED** [review]
LGTM

**Review by markpit — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (mwestacott) [reply README.md:16]: Yes, I think whenever there's a commit and something needs documenting in the README related to it, that's the time to update the file.

File: Makefile (hunk 1 above)
Comment (mwestacott) [reply Makefile:19]: They will be needed but they won't work until those other tasks are delivered, so I'd essentially be delivering a Makefile with non-working targets. I think I'd prefer to leave them commented out with a general qualifying comment.

**Review by dlloyd1-ch — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (mwestacott) [reply Makefile:22]: Done.

**Review by dlloyd1-ch — APPROVED** [review]

**Pull request merged**